torch/csrc/jit/tensorexpr/kernel.cpp

```diff
@@ -941,6 +941,7 @@ Tensor TensorExprKernel::bindInput(const torch::jit::Value* input) {
           "t" + input_name_map_[input],
           {0},
```

Comment:
to handle this case (where buffer size is set to 0) I added the is_flattened_ variable; is there a better way to handle this?

Reply:
How about the case of non-contiguous tensors, e.g. 1-dimensional tensors with stride that is not 1? In that case, how could we tell that we are handling the special case of a 1d flattened buffer?

Reply:
If I remember correctly, NNC will convert dis-contiguous tensors to contiguous buffers before generating code. That is, all buffers inside NNC are contiguous but this line : https://github.com/pytorch/pytorch/blob/3b27304d20f53619344b74e683357424ca3d4029/torch/csrc/jit/tensorexpr/kernel.cpp#L958
We can skip bound checks for dis-contiguous tensors for now since there are no such buffer accesses. @ZolotukhinM correct me if I'm wrong.
  
For flattened buffers, we do need to make sure that there are no out of bound buffer accesses.

Reply:
btw, the size of discontiguous tensors are not computed, and we are using size 0 for `inBuffer`. This will trigger an error in your bound check. I am fixing it in this PR: https://github.com/pytorch/pytorch/pull/69657

Reply:
thanks! this addresses my concern about the 1D tensors with 1D flattened buffer